### ISSUE TITLE
Support x-nullable as an alternative to nullable behind an option

### ISSUE THREAD

**Issue by mitchell-merry** [kubb-project/kubb#846]
### What is the problem this feature would solve?

In Swagger 2.0, there is no way to represent that a type is "nullable", i.e., that `null` is a valid value to pass to it (like `nullable` in Openapi 3.0.x).

### External documents/projects?

No mention of `nullable` in the 2.0 spec: https://swagger.io/specification/v2/

`nullable: true` is how it is done in 3.0.x: https://swagger.io/docs/specification/data-models/data-types#null

(3.1.x uses an `anyOf` with `null`)

### What is the feature you are proposing to solve the problem?

A common convention is to use `x-nullable` to represent this instead. I would like it if kubb supported this (specifically, for `swagger-zod`).

I recently implemented this in `openapi-typescript`: https://github.com/drwpow/openapi-typescript/pull/1576, and need this feature to maintain feature parity (currently we cannot use these zod schemas as they miss this crucial feature). If approved, I would make the PR to support it here.

The core of the change would effectively be changing [this](https://github.com/kubb-project/kubb/blob/df5da5a546803eb08c158b556edfc2241ae20da7/packages/swagger-zod/src/ZodGenerator.ts#L125):
```ts
const nullable = schema.nullable ?? false
```
to this (but exposed as an option):
```ts
const nullable = (schema.nullable || schema["x-nullable"]) ?? false
```

### What alternatives have you considered?

Upgrading to 3.1.x... but that is impossible for a number of reasons right now 😅

**Comment by stijnvanhulle** [comment]
@mitchell-merry It would of course be nice if this could be fixed in `oas` but I will open a PR that will add the logic you suggested `const nullable = (schema.nullable || schema["x-nullable"]) ?? false` so the issue here will be resolved :)

**Comment by stijnvanhulle** [comment]
Fixed with version `2.7.2`
